routers/web/repo/setting.go

```diff
@@ -1168,8 +1195,25 @@ func DeployKeysPost(ctx *context.Context) {
 	ctx.Redirect(ctx.Repo.RepoLink + "/settings/keys")
 }
 
+func DeleteSecret(ctx *context.Context) {
+	if err := secret_service.DeleteSecretByID(ctx, ctx.FormInt64("id")); err != nil {
+		ctx.Flash.Error("DeleteSecretByID: " + err.Error())
```

Comment:
Erm… You are displaying an error as a log message in the front end?
Why not a 
```suggestion
		ctx.Flash.Error(ctx.Tr("repo.settings.secret_deletion_error", error))
```
for a new translation string?

Reply:
Additionally, you should also log the error in the backend.
Or are you explicitly omitting that log to not give any hints to whoever gets access to the logs?

Reply:
Solved.